util/util.py

```diff
@@ -0,0 +1,33 @@
+import collections.abc
+
+def listify( val, require_string=False ):
+    """Return a list version of val.
+
+    If val is already a sequence other than a string, return list(val).
+    Otherwise, return [val].  If val is None, return None.
+
+    Parameters
+    ----------
+    require_string: bool (default False)
+       If true, then val must either be a sequence of strings or a string
+
+    Returns
+    -------
+    list or None
+
+    """
+
+    if val is None:
+        return val
+
+    if isinstance( val, collections.abc.Sequence ):
+        if isinstance( val, str ):
+            return [ val ]
+        else:
+            if require_string and ( not all( [ isinstance( i, str ) for i in val ] ) ):
+                raise TypeError( f'listify: all elements of passed sequence must be strings.' )
+            return list( val )
+    else:
+        if require_string and ( not isinstance( val, str ) ):
```

Comment:
I think both "and not" statements could do without parenthesis, but it is just a matter of style so I leave it up to you.